Saturation/ConsequenceFinder.hpp

```diff
@@ -27,6 +27,10 @@
 
 #include "Inferences/TautologyDeletionISE.hpp"
 
+namespace Lib {
```

Comment:
Generally in favour of local forward declarations, where necessary, but when one gets longer than a oneliner ... wouldn't 

`#include "Lib/SkipList.hpp"`

be easy to understand?

(In this particular case, by the way, the SkipList seems to be coming from some nested include anyway...)

Reply:
Sure, depends on the tradeoff between "touched one header, recompile everything" and ease of maintenance. Let me know what you want to do in general, this one in particular can just go, as you say.